move substitute lib to open source

linbinyu opened the pull request:
Summary: as title

Test Plan: buck build //xplat/caffe2:generated_aten_config_header

Reviewed By: kit1980

Differential Revision: D36989348

malfet commented on tools/substitute.py:
```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
```
Those looks like artifacts from Python-2 era
```suggestion
```

malfet commented on tools/substitute.py:
```diff
@@ -0,0 +1,29 @@
+        contents = f.read()
+
+    output_file = os.path.join(options.install_dir, options.output_file)
+    try:
+        os.makedirs(os.path.dirname(output_file))
+    except OSError:
+        pass
```
Wouldn't `exists_ok` be the right solution here (again, present only in Python-3.2+)
```suggestion
     os.makedirs(os.path.dirname(output_file), exists_ok=True)
```

linbinyu replied:
I just copied this file. I will do what you said and see if it works.

malfet reviewed (approved):
LGTM, but can we make it a bit more Python-3 friendly?

linbinyu committed:
move substitute lib to open source (#79093)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/79093

as title

Test Plan: buck build //xplat/caffe2:generated_aten_config_header

Reviewed By: kit1980

Differential Revision: D36989348

fbshipit-source-id: 09ae786d307fe239a3e3bd1d6aa6de285a808d88
